builder/parser/line_parsers.go

```diff
@@ -233,7 +233,7 @@ func parseString(rest string) (*Node, map[string]bool, error) {
 // parseJSON converts JSON arrays to an AST.
 func parseJSON(rest string) (*Node, map[string]bool, error) {
 	var myJson []interface{}
-	if err := json.Unmarshal([]byte(rest), &myJson); err != nil {
+	if err := json.NewDecoder(strings.NewReader(rest)).Decode(&myJson); err != nil {
```

Comment:
there's no need for this. It's better to use an io.Reader only when the content is not all in memory. In this case it's already in memory so I don't see why not keep Unmarshal.